### PR TITLE
[MBL-16264][Student] Share Extension | Retry upload from progress screen

### PR DESCRIPTION
**Pull request by kristofnemere**
Test plan: smoke test share extension uploads, and other file uploads using the file upload dialog (teacher speedgrader comments too)

refs: MBL-16264
affects: Student
release note: Added proper failed screen and retry logic for share extension.

## Checklist

- [x] Follow-up e2e test ticket created or not needed
- [x] Run E2E test suite or not needed
- [x] Tested in dark mode
- [x] Tested in light mode
- [x] A11y checked
- [x] Approve from product or not needed

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: libs/pandautils/src/test/java/com/instructure/pandautils/features/shareextension/progress/ShareExtensionProgressViewModelTest.kt [hunk 1]
```diff
@@ -251,13 +266,151 @@ class ShareExtensionProgressViewModelTest {
...
+        }
+
+        assertEquals("Submission", viewData?.dialogTitle)
+        assertEquals("Error", viewData?.subtitle)
+        assertEquals(true, viewData?.failed)
+    }
+
+    @Test
+    fun `Failed upload retry`() {
+        every { resources.getString(R.string.fileUploadFailedSubtitle) } returns "Error"
```
Comment (hermannakos) [inline libs/pandautils/src/test/java/com/instructure/pandautils/features/shareextension/progress/ShareExtensionProgressViewModelTest.kt:331]: This should be in the `setupString()` with the other string resources.

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

File: libs/pandautils/src/test/java/com/instructure/pandautils/features/shareextension/progress/ShareExtensionProgressViewModelTest.kt [hunk 2]
```diff
@@ -251,13 +266,151 @@ class ShareExtensionProgressViewModelTest {
         assertEquals("50.0%", viewData?.percentage)
     }
 
+    @Test
+    fun `Failed upload maps correctly`() {
+        every { resources.getString(R.string.fileUploadFailedSubtitle) } returns "Error"
```
Comment (hermannakos) [inline libs/pandautils/src/test/java/com/instructure/pandautils/features/shareextension/progress/ShareExtensionProgressViewModelTest.kt:271]: This should be in the `setupString()` with the other string resources.

File: libs/pandautils/src/test/java/com/instructure/pandautils/features/shareextension/progress/ShareExtensionProgressViewModelTest.kt [hunk 3]
```diff
@@ -70,12 +78,19 @@ class ShareExtensionProgressViewModelTest {
     }
 
     @Test
-    fun `Show error dialog when upload failed`() {
+    fun `Show error on progress dialog when upload failed`() {
+        every { resources.getString(R.string.fileUploadFailedSubtitle) } returns "Error"
```
Comment (hermannakos) [inline libs/pandautils/src/test/java/com/instructure/pandautils/features/shareextension/progress/ShareExtensionProgressViewModelTest.kt:82]: This should be in the `setupString()` with the other string resources.

**Review by tamaskozmer — APPROVED** [review]
QA + 1

**Review by hermannakos — CHANGES_REQUESTED** [review]

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by hermannakos — APPROVED** [review]

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by kdeakinstructure — APPROVED** [review]
QA 👍

**Pull request merged**